ComSemApp/templates/ComSemApp/teacher/course_students_detail.html

```diff
@@ -0,0 +1,32 @@
+{% load static %}
+{% if course.students.all %}
+<thead>
+	<tr>
+		<th>Name</th>
+		<th>Worksheets Assigned</th>
+		<th>Ungraded Submissions</th>
+		<th>Number of Attempts</th>
+
+	</tr>
+</thead>
+
+<tbody>
+	{% for student in course.students.all %}
+	<tr>
+		<td>{{student}}</td>
+		<td> 			<!-- using username to get the calculated count for submissions for that student in this class section -->
+			{{ worksheets|get_item:student.user.username }}</td>
```

Comment:
With an annotate I think we wouldn't have to access these three secondary dictionaries. 
It would be a bit more straightforward to loop through a list of students provided by the view, and to be able to access the data directly in the student object `student.worksheet_count`.